packages/common/src/api/tan-query/wallets/useTokenBalance.ts

```diff
@@ -0,0 +1,145 @@
+import { useCallback } from 'react'
+
+import { FixedDecimal } from '@audius/fixed-decimal'
+import { TokenAccountNotFoundError } from '@solana/spl-token'
+import { Commitment } from '@solana/web3.js'
+import { useQuery, useQueryClient } from '@tanstack/react-query'
+
+import { useCurrentAccountUser } from '~/api'
+import { useQueryContext } from '~/api/tan-query/utils'
+import { Status } from '~/models/Status'
+import { MintName } from '~/services/audius-backend/solana'
+import { getUserbankAccountInfo, getTokenBySymbol } from '~/services/index'
+
+import { QUERY_KEYS } from '../queryKeys'
+import { QueryOptions, type QueryKey } from '../types'
+
+const createTokenBalance = (
```

Comment:
nit: is this rly the cleanest way? can we not do something like:

`return !!amount ? new FixedDecimal(BigInt(amount.toString()), decimals) : null`

Reply:
idk i guess that's kind of annoying too..